### ISSUE TITLE
Use SQLCommandTimeout in BaseDataProvider

### ISSUE THREAD

**Issue by Manuel-Innovapps** [nopSolutions/nopCommerce#6271]
nopCommerce version: 4.50.x +

Steps to reproduce the problem: 
1. We migrated our shops from onPrem to Azure and hat timeout problems with the Bestseller Report in the Front & Backend.
2. It was throwing us error about timeouts, so as a first step, we naturally increased the timeout in the appsettings.
3. As far as we could tell, the timeout wasn't changing anything for selects, which seemed to be necessary for us.
4. Once we applied this changed, things were really slow on loading, but at least they worked.

We propose to implement this fix to make the SQLCommand available consistently between all SQL Commands (Insert, delete, select etc.)

Current Implementation:
https://github.com/nopSolutions/nopCommerce/blob/804a1a97732fc01574c88ea8b37b9f0fc48ce9fd/src/Libraries/Nop.Data/DataProviders/BaseDataProvider.cs#L262-L266

Proposed Implementation:
```c#
        public virtual IQueryable<TEntity> GetTable<TEntity>() where TEntity : BaseEntity
        {
            return new DataContext(LinqToDbDataProvider, GetCurrentConnectionString()) { 
                    MappingSchema = GetMappingSchema(), 
                    CommandTimeout = DataSettingsManager.GetSqlCommandTimeout()
                }.GetTable<TEntity>();
        }
```

**Comment by skoshelev** [comment]
Hi @Manuel-Innovapps. Thanks for the suggestion, we will implement it. You can also look at the solution to ticket #5792 that improves the performance of this report.

**Comment by exileDev** [comment]
Thanks. Closed #6271
